pkg/ocm/billing.go

```diff
@@ -0,0 +1,58 @@
+package ocm
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/rosa/pkg/helper"
+)
+
+var ROSAHypershiftQuota = "cluster|byoc|moa|marketplace"
+
+var awsAccountRegexp = regexp.MustCompile(`^[0-9]{12}$`)
+
+func (c *Client) GetBillingAccounts() ([]string, error) {
+	acctResponse, err := c.ocm.AccountsMgmt().V1().CurrentAccount().
+		Get().
+		Send()
+	if err != nil {
+		return nil, handleErr(acctResponse.Error(), err)
+	}
+	organization := acctResponse.Body().Organization().ID()
+	search := fmt.Sprintf("quota_id='%s'", ROSAHypershiftQuota)
+	quotaCostResponse, err := c.ocm.AccountsMgmt().V1().Organizations().
+		Organization(organization).
+		QuotaCost().
+		List().
+		Parameter("fetchCloudAccounts", true).
+		Parameter("search", search).
+		Page(1).
+		Size(-1).
+		Send()
+	if err != nil {
+		return nil, handleErr(quotaCostResponse.Error(), err)
+	}
+
+	var billingAccounts []string
+	for _, item := range quotaCostResponse.Items().Slice() {
+		for _, account := range item.CloudAccounts() {
+			if account.CloudProviderID() == "aws" && !helper.Contains(billingAccounts, account.CloudAccountID()) {
+				billingAccounts = append(billingAccounts, account.CloudAccountID())
+			}
+		}
+	}
+
+	if len(billingAccounts) == 0 {
+		return billingAccounts, errors.New("no billing accounts found")
+	}
+
+	return billingAccounts, nil
+}
+
+func IsValidAWSAccount(account string) bool {
+	if len(account) == 0 || awsAccountRegexp.MatchString(account) {
```

Comment:
Maybe we should call this function only if `account != ""`?
It might be confusing that we consider here an empty string as a valid AWS account.

Reply:
My original idea was to keep the len(account) == 0 only until this attribute becomes mandatory (which will).

But I see how this can be confusing, as you pointed out. I will move this logic out of this function.